src/AdminConsole/EventLog/Loggers/EventLoggerEfReadStorage.cs

```diff
@@ -4,26 +4,34 @@
 
 namespace Passwordless.AdminConsole.EventLog.Loggers;
 
-public class EventLoggerEfReadStorage : IEventLoggerStorage
+public class EventLoggerEfReadStorage<TDbContext> : IEventLoggerStorage where TDbContext : ConsoleDbContext
```

Comment:
Why do we need this change? Couldn't we still do `IDbContextFactory<ConsoleDbContext>` if we registered the corresponding abstraction?

Reply:
No because it cannot resolve ConsoleDbContext. We tried to use AddTransient<ConsoleDbContext, SqliteDbContext>() for example. But it still won't work.

With AddDbContext you can do AddDbContext<ConsoleDbContext, SqliteDbContext>(). But not when you use the factory.

Reply:
Ok, makes sense